File: backend/src/main/java/es/udc/paproject/backend/model/services/GameService.java
package es.udc.paproject.backend.model.services;

import java.time.LocalDateTime;
import java.util.List;

import es.udc.paproject.backend.model.entities.Game;
import es.udc.paproject.backend.model.exceptions.InstanceNotFoundException;
import es.udc.paproject.backend.model.exceptions.StartDateAfterEndDateException;

public interface GameService {
    
    Game addGame(Long teamId, Long seasonId, LocalDateTime gameDate, String rival, String description) throws InstanceNotFoundException;

    //solo añado el obj jugador y el partido a la tabla playerGame
    void addPlayerToGame(Long gameId, Long playerId) throws InstanceNotFoundException;

    Game findGameById(Long gameId) throws InstanceNotFoundException;

    List<Game> findGamesByPlayerId(Long playerId) throws InstanceNotFoundException;
    
    List<Game> findGamesByUserId(Long userId) throws InstanceNotFoundException;

    List<Game> findGamesByTwoDatesAndTeamIdOrSeasonId(Long userId, Long teamId, Long seasonId, LocalDateTime startDate, LocalDateTime endDate) throws StartDateAfterEndDateException, InstanceNotFoundException;

    List<Game> findGamesByTeamId(Long teamId) throws InstanceNotFoundException;

    List<Game> findGamesBySeasonId(Long seasonId) throws InstanceNotFoundException;

    //no pasar al front. Usar las dos anteriores
    List<Game> findGamesByTeamIdAndSeasonId(Long teamId, Long seasonId) throws InstanceNotFoundException;

    void removeGame(Long gameId) throws InstanceNotFoundException;

    void removePlayerToGame(Long playerId, Long gameId) throws InstanceNotFoundException;

    Game updateGame(Long gameId, LocalDateTime gameDate, String rival, String description) throws InstanceNotFoundException;
}

File: frontend/src/backend/gameService.js
import {config, appFetch} from './appFetch';

export const findGameById = (gameId, onSuccess, onErrors) =>
    appFetch(`/games/${gameId}`, config('GET'), onSuccess, onErrors);

export const findGamesByUserId = (onSuccess, onErrors) =>
    appFetch('/games/user', config('GET'), onSuccess, onErrors);

export const findGamesByPlayerId = (playerId, onSuccess, onErrors) =>
    appFetch(`/games/${playerId}/player`, config('GET'), onSuccess, onErrors);

export const findGamesByTwoDatesAndTeamIdOrSeasonId = (teamId, seasonId, startDate, endDate, onSuccess, onErrors) =>
    appFetch(`/games/dates?teamId=${teamId}&seasonId=${seasonId}&startDate=${startDate}&endDate=${endDate}`, config('GET'), onSuccess, onErrors);

export const findGamesByTwoDates= (startDate, endDate, onSuccess, onErrors) =>
    appFetch(`/games/dates?startDate=${startDate}&endDate=${endDate}`, config('GET'), onSuccess, onErrors);

export const findGamesByTwoDatesAndTeamId = (teamId, startDate, endDate, onSuccess, onErrors) =>
    appFetch(`/games/dates?teamId=${teamId}&startDate=${startDate}&endDate=${endDate}`, config('GET'), onSuccess, onErrors);

export const findGamesByTwoDatesAndSeasonId = (seasonId, startDate, endDate, onSuccess, onErrors) =>
    appFetch(`/games/dates?seasonId=${seasonId}&startDate=${startDate}&endDate=${endDate}`, config('GET'), onSuccess, onErrors);

export const findGamesByTeamId = (teamId, onSuccess, onErrors) =>
    appFetch(`/games/${teamId}/team`, config('GET'), onSuccess, onErrors);

export const findGamesBySeasonId = (seasonId, onSuccess, onErrors) =>
    appFetch(`/games/${seasonId}/season`, config('GET'), onSuccess, onErrors);

export const addGame = (teamId, seasonId, gameDate, rival, description, onSuccess, onErrors) =>
    appFetch(`/games?teamId=${teamId}&seasonId=${seasonId}&gameDate=${gameDate}&rival=${rival}&description=${description}`, config('POST'), onSuccess, onErrors);

export const addGameWithTeam = (teamId, gameDate, rival, description, onSuccess, onErrors) =>
    appFetch(`/games?teamId=${teamId}&gameDate=${gameDate}&rival=${rival}&description=${description}`, config('POST'), onSuccess, onErrors);

export const addGameWithSeason = (seasonId, gameDate, rival, description, onSuccess, onErrors) =>
    appFetch(`/games?seasonId=${seasonId}&gameDate=${gameDate}&rival=${rival}&description=${description}`, config('POST'), onSuccess, onErrors);


export const addPlayerToGame = (playerId, gameId, onSuccess, onErrors) =>
    appFetch(`/games/${playerId}/addPlayerToGame?gameId=${gameId}`, config('POST'), onSuccess, onErrors);

export const updateGame = (gameId, gameDate, rival, onSuccess, onErrors) =>
    appFetch(`/games/${gameId}?gameDate=${gameDate}&rival=${rival}`, config('PUT'), onSuccess, onErrors);

export const removeGame = (gameId, onSuccess, onErrors) =>
    appFetch(`/games/${gameId}`, config('DELETE'), onSuccess, onErrors);

    export const removePlayerToGame = (playerId, gameId, onSuccess, onErrors) =>
    appFetch(`/games/${playerId}/player?gameId=${gameId}`, config('DELETE'), onSuccess, onErrors);